frontend/model/state.js

```diff
@@ -169,7 +169,11 @@ const getters = {
     // due to the same flow issue as https://github.com/facebook/flow/issues/5838
     return Object.keys(contracts)
       .filter(contractID => contracts[contractID].type === 'group' && state[contractID].settings)
-      .map(contractID => ({ groupName: state[contractID].settings.groupName, contractID }))
+      .map(contractID => ({
+        groupName: state[contractID].settings.groupName,
+        contractID,
+        groupPicture: state[contractID].settings.groupPicture
```

Comment:
So the reason why this getter returned just `groupName` and `contractID` is because with that information, you can get everything else you need for each group.
```suggestion
        groupPicture: state[contractID].settings.groupPicture
```
```suggestion
        groupPicture: state[contractID].settings.groupPicture
```

This function is not designed to return everything you might want about all groups. It is designed as a hook into knowledge about all groups, that can be used as such to the degree required.

So there is a danger her of unnecessarily continuing to overload this function with more and more information that is not needed 99% of the time it is used. That is creates a problem because this function is used in many places, and so we would be repeatedly constructing massive objects and passing around that data to places that don't ever make use of that data. In a word, it is wasteful.

So I strongly recommend that instead of taking the approach of adding `groupPicture` to this returned object, that you instead use `group.contractID` (in GroupsList.vue below) to retrieve the data you need.

So this getter would be restored to how it was, and down below it would read something like `$store.state[group.contractID].settings.groupPicture`